### PR TITLE
Adding test, updating test one_to_one

### PR DESCRIPTION
**Pull request by soumilbaldota**
- uncommenting assertions in vector_members test
- testing more than one contribution point to one MCParticle
- removing additional includes

### CODE REVIEW

File: test/one_to_many.jl [hunk 1]
```diff
@@ -91,15 +89,15 @@ function read_data(a::MCParticleCollection)
     @assert mcp3.generatorStatus ==3 
     @assert length(mcp3.parents)== 1
     @assert mcp3.parents[1] == mcp1
-#     @assert mcp3.parents == MCParticleCollection([mcp1])
+    @assert mcp3.parents == MCParticleCollection([mcp1])
```
Comment (tmadlener) [inline test/one_to_many.jl]: As you clarified in the email. These are indeed redundant if you check the contents explicitly above.

File: test/one_to_one.jl [hunk 2]
```diff
@@ -71,10 +65,13 @@ function read_data(a::CaloHitContributionCollection)
 	@assert sccon2.stepPosition == Vector3d{Float32}(7, 6.32, 12)
 	@assert sccon2.particle.PDG == 2212
 	@assert sccon2.particle.mass == 0.938
-	@assert sccon2.particle.momentumAtEndpoint == Vector3d{Float32}(0.0,0.0,-7000.0)
+	@assert sccon2.particle.momentumAtEndpoint == Vector3d{Float32}(0.0,0.0,7000.0)
 	@assert sccon2.particle.generatorStatus == 3
 	@assert sccon2.energy == Float32(13.6)
 
+	# checking egality of particle object
+	@assert sccon2.particle === sccon1.particle
```
Comment (tmadlener) [inline test/one_to_one.jl:73]: This essentially already encompasses what we want to know. It is not yet entirely clear to me if we can manage this in the templates as well. Could you add a few explicit checks that compare a few properties of the particle with just the equality operator, i.e. something like
```julia
@assert sccon2.particle.PDG == sccon1.particle.PDG
```

File: test/true_one_to_many.jl [hunk 3]
```diff
@@ -0,0 +1,50 @@
+include("../MCParticle.jl")
+
+function test_copy()
```
Comment (tmadlener) [inline test/true_one_to_many.jl]: I am not sure if `test_copy` is the best name in this case, as we are testing relations (in podio speak), resp. references here.

File: test/true_one_to_many.jl [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+    mcp1.mass = 0.612
+    mcp2.mass = 0.745
+
+    # passes if values are changed in parents 
+
+    @assert mcp3.parents[1].mass == 0.612
+    @assert mcp4.parents[1].mass == 0.745
+    @assert mcp5.parents[1].mass == 0.612
+    @assert mcp5.parents[2].mass == 0.745
```
Comment (tmadlener) [inline test/true_one_to_many.jl:47]: :+1:

File: test/true_one_to_one.jl [hunk 5]
```diff
@@ -0,0 +1,36 @@
+include("../CaloHitContribution.jl")
+
+function test_copy()
+
+	mcp1 = MCParticle()
+    mcp1.PDG = 2212
```
Comment (tmadlener) [inline test/true_one_to_one.jl]: There seem to be some whitespace differences here for some of the lines (tabs vs. spaces most likely)

**Review by tmadlener — COMMENTED** [review]
I think this looks pretty good, and I also think we cover the major use cases. We will most likely discover more things to test along the way though.

**Comment by soumilbaldota** [comment]
@tmadlener this update fixes all the above.

**Pull request merged**